ravenframework/SupervisedLearning/DynamicModeDecompositionControl.py

```diff
@@ -225,30 +241,72 @@ def __trainLocal__(self,featureVals,targetVals):
-      This method is used to inquire the DMD to evaluate (after normalization that in
-      this case is not performed)  a set of points contained in featureVals.
-      a KDTree algorithm is used to construct a weighting function for the reconstructed space
-      @ In, featureVals, numpy.ndarray, shape= (n_requests, n_timeStep, n_dimensions), an array of input data
-      @ Out, returnEvaluation , dict, dictionary of values for each target (and pivot parameter)
+      Method to return the features' importances
+      @ In, None
+      @ Out, importances, dict , dict of importances {feature1:(importanceTarget1,importanceTarget2,...),
+                                                              feature2:(importanceTarget1,importanceTarget2,...),...}
     """
+    if self._importances is None:
+      from sklearn import preprocessing
+      from sklearn.ensemble import RandomForestRegressor
+      # the importances are evaluated in the transformed space
+      importanceMatrix = np.zeros(self.__Ctilde.shape)
+      for smp in range(self.__Ctilde.shape[0]):
+        importanceMatrix[smp,:,:] = self.__Ctilde[smp,:,:]
+        scaler = preprocessing.MinMaxScaler()
+        scaler.fit(importanceMatrix[smp,:,:].T)
+        importanceMatrix[smp,:,:] = scaler.transform(importanceMatrix[smp,:,:].T).T
+
+      self._importances = dict.fromkeys(self.parametersIDs+self.stateID,1.)
+
+      # the importances for the state variables are inferred from the C matrix/operator since
+      # directely linked to the output variables
+      minVal, minIdx = np.finfo(float).max, -1
+      for stateCnt, stateID in enumerate(self.stateID):
+        # for all outputs
+        self._importances[stateID] = np.asarray([abs(float(np.average(importanceMatrix[:,outcnt,stateCnt]))) for outcnt in range(len(self.outputID))])
+        if minVal > np.min(self._importances[stateID]):
+          minVal = np.min(self._importances[stateID])
+          minIdx = stateCnt
+      # as first approximation we assume that the feature importance
+      # are assessable via a perturbation of the only feature space
+      # on the C matrix
+      for featCnt, feat in enumerate(self.parametersIDs):
+        permutations = set(self.parameterValues[:,featCnt])
+        indices = [np.where(self.parameterValues[:,featCnt] == elm )[-1][-1]  for elm in permutations]
+        self._importances[feat] = np.asarray([abs(float(np.average(importanceMatrix[indices,outcnt,minIdx]))) for outcnt in range(len(self.outputID))])
+      self._importances = dict(sorted(self._importances.items(), key=lambda item: np.average(item[1]), reverse=True))
+
+    if group is not None:
+      groupMask = np.zeros(len(self.outputID),dtype=bool)
+      for cnt, oid in enumerate(self.outputID):
+        if oid in group:
+          groupMask[cnt] = True
+        else:
+          groupMask[cnt] = False
+      newImportances  = {}
+      for key in self._importances:
+        newImportances[key] =  newImportances[key][groupMask]
```

Comment:
newImportances is an empty dict, do you mean "newImportances[key] =  self._importances[key][groupMask]"

Reply:
ops. Fixed